cmd/root.go

```diff
@@ -97,18 +98,40 @@ func RunCmdRoot() {
 
 	targetOptions, err := opts.GetWarmupTargetOptions()
 
+	result := warmup.Result{TotalRequests: 0, ClientErrors: 0, RequestsSent: 0}
 	if err == nil {
 		wp, err1 := createWarmup(targetOptions, done)
 		if err1 == nil {
-			runWarmup(wp, done)
+			result = runWarmup(wp, done)
 		}
 	}
+	log.Printf("summary: total_requests: %d, requests_sent: %d, bad_requests: %d", result.TotalRequests, result.RequestsSent, result.ClientErrors)
 
 	if opts.ServerProbe.Enabled {
-		probeServer.IsReady(true)
+		if opts.FailReadiness.Enabled {
+			if opts.FailReadiness.NoRequests && result.RequestsSent == 0 {
+				log.Print("probeServer readiness failed, no requests sent")
+			} else if opts.FailReadiness.ClientErrors && result.ClientErrors > 0 {
```

Comment:
this seems a dangerous option and something that can lead to incidents.
i think it's normal if some requests fail with 4xx errors, specially with the random placeholders.

Reply:
ps: i know i suggested this :(

Reply:
i wonder if we could solve this differently.
i think the main issue that we have right now is with api drift: we configure mittens once, but then you modify endpoints, and add/delete query params, etc, and after some time the warmup isn't warming up anything.

adding a lot of options makes this complicated to use (which it already is based on feedback that we got).
i would try to have just a single flag: "fail if warmup didn't run".
the api drift could be solved with logs/metrics, but to be honest, if folks already have a logging/metrics system today they would see the drift even without explicit metrics from Mittens.

Reply:
Indeed. I also don't think it's a good idea to fail in case of 4XXs.

I've refactored the PR and only exposing a single flag, `fail-readiness`, which will fail Mittens readiness only when no requests are sent.

I believe that's useful. At least users can make sure that it ran and didn't error before sending requests.